src/convertToObject.js

```diff
@@ -1,18 +1,15 @@
 'use strict';
 
-/**
- * Implement convertToObject function:
- *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
- * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
- *
- * @param {string} sourceString
- *
- * @return {object}
- */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .filter(property => property.includes(':'))
+    .map(property => property.split(':').map(word => word.trim()))
+    .reduce((key, value) => {
+      key[value[0]] = value[1];
```

Comment:
A lot of nested iterations, while we can do it in just one iteration. Let's optimize algorythm productivity.